### ISSUE TITLE
GetNextPage() for ListContentItemsAsync is throwing an NotFound error

### ISSUE THREAD

**Issue by twadevs** [Kentico/kontent-management-sdk-net#158]
Hi,

We are currently using the management v1 SDK for various migration tasks - but just noticed that as of today we can get the next page for the ListContentItemsAsync  response. This is the error:

**Unknown error. HTTP status code: NotFound. Reason phrase: Not Found.; Kentico.Kontent.Management;    at Kentico.Kontent.Management.Modules.HttpClient.ManagementHttpClient.SendAsync(IMessageCreator messageCreator, String endpointUrl, HttpMethod method, HttpContent requestContent)\r\n   at Kentico.Kontent.Management.Modules.ActionInvoker.ActionInvoker.InvokeReadOnlyMethodAsync[TResponse](String endpointUrl, HttpMethod method)\r\n   at Kentico.Kontent.Management.ManagementClient.GetNextItemsListingPageAsync(String continuationToken)\r\n   at Kentico.Kontent.Management.Models.ListingResponseModel`1.GetNextPage()**

And the code we are using:

` if (listResponse.HasNextPage())
                {
                    listResponse = await listResponse.GetNextPage();                    
                }
                else
                {
                    continueDelete = false;
                }`

**Comment by IvanKiral** [comment]
Hello,

thank you for letting us know about your issue.

Unfortunately, I could not replicate your error, could you please provide us with more context? Also could you try whether the same error occur when you use Management SDK v2? What would it take you to use v2?

**Comment by twadevs** [comment]
Hi,

Thanks for your response.

Just wanted to give an update:

I got around the issue by skipping the GetNextPage() call and getting a new set of 100 items using ListContentItemsAsync. We are using this script to clear out the project to do a fresh migration, as part of testing. So the code at the moment gets 100 items, deletes them, and then gets another 100.

```
 ListingResponseModel<ContentItemModel> listResponse = await _settings.Client.ListContentItemsAsync();

            var continueDelete = true;

            await _settings.ElasticContext.AddLogMessage($"DELETING: Starting deleting all items - (Current page count {listResponse.Count()})");

            while (continueDelete)
            {

                foreach (var item in listResponse)
                {
                    if (item.CodeName != RootContainer.Codename)
                    {
                        var identifier = ContentItemIdentifier.ByCodename(item.CodeName);
                        await _settings.Client.DeleteContentItemAsync(identifier);
                        await _settings.ElasticContext.AddLogMessage($"DELETING: Deleted {item.Type}-{item.CodeName}");
                    }
                }

                if (listResponse.HasNextPage())
                {
                    // listResponse = await listResponse.GetNextPage();
                    listResponse = await _settings.Client.ListContentItemsAsync();
                    await _settings.ElasticContext.AddLogMessage($"DELETING: Found more items to delete. Count: {listResponse.Count()})");
                }
                else
                {
                    continueDelete = false;
                }

            }
```

We are using SDK v1 because the v2 SDK wasn't available yet. Since this is only a once-off migration, it won't be worth switching to v2, but we'll look into it for our next project  :) 

Thanks,

**Comment by gormal** [comment]
Thanks for the update!

Happy to hear that you plan to use the new version of SDK for your next update.

**Comment by gormal** [comment]
@mbridges-envative could you please try to use version [2.2.0-beta1](https://github.com/Kentico/kontent-management-sdk-net/releases/tag/2.2.0-beta1)?

**Comment by mbridges-envative** [comment]
@gormal would you be able to provide any details to how/why this issue arose while using the latest, official Management SDK release (2.1.0)?

**Comment by mbridges-envative** [comment]
@gormal thanks for providing this feedback. I'm just curious why this breaking change would have been introduced, but not supported by an official SDK release.

**Comment by gormal** [comment]
Well, to be honest, we didn't invest much time in .NET SDK previously and when that has changed we immediately start working on a new major version that would support all endpoints from API v2 which is about to be released really soon.

**Comment by gormal** [comment]
yeah, that would be great! 3.0.0-beta7 is the last beta before the major release, and we expect that version 3.0.0 should be basically the same as 3.0.0-beta7.
